### DISCUSSION TITLE
SvelteKit Form Actions not Working

### DISCUSSION THREAD

**Discussion by DanielRios549** [vercel/community#4460] [Bug Report]
### Summary

I have an error with named actions on Vercel, first of all, I'm using only named actions, and all actions work in development and preview, the problem is when I deploy Vercel, all actions throw error 500 and the logs always say "No action with name 'default'", but I'm not using them as I said.

I simply make a POST request to `/route?/action` and the log says "No action with name 'default'".

![image](https://github.com/vercel/community/assets/11152535/09345377-e973-48b6-800b-9bf92639f5c2)

### Steps to Reproduce

Create a SvelteKit project and use a named function explicitly in the form action: `/route?/action`, it will work locally but not after deploy to Vercel.

**Comment by DanielRios549** [comment]
This commit in my project "fixes" this problem, it does not fix the problem with named form actions, but the default actions work without issue.

https://github.com/DanielRios549/DanielRiosDev/commit/eefb4dcea7389c093c12f50bf51bbffe607b11d7

**Reply by amyegan** [reply]
I'm glad you found a workaround. I wasn't able to replicate the problem with a fresh SvelteKit template. Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) that we can use for debugging?

**Reply by DanielRios549** [reply]
Ok, I will create a fresh project to replicate this problem

**Reply by DanielRios549** [reply]
@amyegan This is weird, I tried to create a minimal reproducible example, but it simply just works!

I use the same configuration, the same `adapter-vercel` and `sveltekit` versions, the same Node version, a custom domain, everything, but in the MRE the named actions simply work.

Somehow, before this commit (https://github.com/DanielRios549/DanielRiosDev/commit/eefb4dcea7389c093c12f50bf51bbffe607b11d7) in this repository, the named actions do not work. I don't know what causes this, a specific package version, a configuration of that package, I simply don't know.

But again, using a fresh SvelteKit application, named actions do work, this problem is specific to that repository, and I think I will never discover why, but is everything ok.

**Reply by amyegan** [reply]
Thanks for giving it a try. Please let me know if you ever see this happen in another project. 🙏

**Comment by carstenjaksch** [comment]
@amyegan I have the same issue and pinned it down to this:

**node_modules/@sveltejs/kit/src/runtime/server/page/actions.js**

```js
async function call_action(event, actions) {
	const url = new URL(event.request.url);

	let name = 'default';
	for (const param of url.searchParams) {
		if (param[0].startsWith('/')) {
			name = param[0].slice(1);
			if (name === 'default') {
				throw new Error('Cannot use reserved action name "default"');
			}
			break;
		}
	}

	const action = actions[name];
	if (!action) {
		throw new Error(`No action with name '${name}' found`);
	}

	if (!is_form_content_type(event.request)) {
		throw new Error(
			`Actions expect form-encoded data (received ${event.request.headers.get('content-type')})`
		);
	}

	return action(event);
}
```

The line `const url = new URL(event.request.url);` tells me this:

```bash
URL {
  href: 'http://localhost:5173/_vercel/speed-insights/vitals',
  origin: 'http://localhost:5173',
  protocol: 'http:',
  username: '',
  password: '',
  host: 'localhost:5173',
  hostname: 'localhost',
  port: '5173',
  pathname: '/_vercel/speed-insights/vitals',
  search: '',
  searchParams: URLSearchParams {},
  hash: ''
}
```

Looks like speed-insights is triggering the default SvelteKit action by accident?

**Reply by carstenjaksch** [reply]
Okay, this is weird. I installed the Vercel CLI locally and tested it. After removing `speed-insights`, it works. But the deployment does not. Same error in Vercel:

```bash
Error: No action with name 'default' found
    at call_action (file:///var/task/.svelte-kit/output/server/index.js:509:11)
    at handle_action_json_request (file:///var/task/.svelte-kit/output/server/index.js:381:24)
    at render_page (file:///var/task/.svelte-kit/output/server/index.js:1887:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async resolve (file:///var/task/.svelte-kit/output/server/index.js:2669:24)
    at async respond (file:///var/task/.svelte-kit/output/server/index.js:2555:22)
    at async Server.default (file:///var/task/.svelte-kit/vercel-tmp/index.js:47:3)
    at async Server.<anonymous> (/var/task/___vc/__launcher/bridge-server-72TT5FOD.js:1:8478)
```

Speed Insights is disabled in Vercel dashboard too.

**Reply by carstenjaksch** [reply]
I found another issue causing this: ISR

https://github.com/sveltejs/kit/issues/11337

Disabling ISR and Speed-Insights together is solving it for me. Hope that gets fixed soon, both are important for my project.
